src/twisted/internet/defer.py

```diff
@@ -1636,36 +1747,41 @@ def loadData(url):
     """
 
     @wraps(f)
-    def unwindGenerator(*args, **kwargs):
+    def unwindGenerator(*args: object, **kwargs: object) -> Deferred:
         try:
             gen = f(*args, **kwargs)
         except _DefGen_Return:
             raise TypeError(
                 "inlineCallbacks requires %r to produce a generator; instead"
                 "caught returnValue being used in a non-generator" % (f,)
             )
-        if not isinstance(gen, types.GeneratorType):
+        # type note: FIXME: error suggests type of gen is wrong:
+        #     Subclass of "Generator[Deferred, object, None]" and "GeneratorType"
+        #     cannot exist: would have incompatible method signatures
+        if not isinstance(gen, GeneratorType):  # type: ignore[unreachable]
```

Comment:
This suggests that `Generator[Deferred, object, None]]` is the wrong return type for `f`.
I could use other eyes on this.